polar-core/src/vm.rs

```diff
@@ -947,6 +953,9 @@ impl PolarVirtualMachine {
             | (Value::RestVariable(l), Value::RestVariable(r)) => {
                 // Two variables.
                 match (self.variable_state(l), self.variable_state(r)) {
+                    // TODO (dhatch): This special case does not seem right to me.
```

Comment:
Agreed.

Reply:
Alex removed in https://github.com/osohq/oso/pull/723/commits/0231cc74c6eaad57b362b40baffbd8026a2153c0